hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java

```diff
@@ -311,7 +314,11 @@ public void persistToken(Token token) {
 						new EntityId(uniqueToken.getOwner()), uniqueToken.getMetadata(), uniqueToken.getCreationTime());
 				currentUniqueTokens.put(merkleUniqueTokenId, merkleUniqueToken);
 				currentUniqueTokenAssociations.associate(new EntityId(uniqueToken.getTokenId()), merkleUniqueTokenId);
-				currentUniqueOwnershipAssociations.associate(treasury, merkleUniqueTokenId);
+				if (uniqueToken.getOwner().equals(token.getTreasury().getId())) {
```

Comment:
When we are minting, we are always minting to the treasury. No need to add the `if`